### PR TITLE
[DO NOT MERGE] WIP general idea how about implementing partial receive in typed

### PR DESCRIPTION
**Pull request by ktoso**
This is just an example for @hseeberger to get inspired how the Actor.partial could be implemented.
There needs to be more type juggling done to get it nice and not expose internals.

It may also break type inference in some places... 

PR is expected to fail currently.

Refs https://github.com/akka/akka/pull/24035

### CODE REVIEW

File: akka-cluster-tools/src/main/scala/akka/cluster/joining/RolloutCallbacks.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package akka.cluster.joining
+
+class RolloutCallbacks {
+
+}
```
Comment (ktoso) [inline akka-cluster-tools/src/main/scala/akka/cluster/joining/RolloutCallbacks.scala]: ignore this

**Comment by ktoso** [comment]
cc @patriknw just FYI

**Comment by hseeberger** [comment]
Thanks! I was guessing this could be implemented like this.

@patriknw I perfer this over your initial suggestion, because it returns a `Behavior` like all other behavior constructors instead of a function. So the API is more consistent. Do you agree?

**Comment by patriknw** [comment]
I see, this is better.

**Comment by ktoso** [comment]
Closing as was just intended to show the impl idea

**Pull request closed without merging**